core/cont/inc/THashList.h

```diff
@@ -36,7 +36,7 @@ class THashTable;
 class THashList : public TList {
 
 protected:
-   THashTable   *fTable;    //Hashtable used for quick lookup of objects
+   THashTable   *fTable;    //! Hashtable used for quick lookup of objects
```

Comment:
Dear Olivier,

> The contributor forgets about the class-version, assuming the added member will be streamed. 

Optimistically one would hope (s)he would have also added some I/O test ...  Also likely the existing test trying to read existing object of that type (assuming there is any such test ...) would start to fail but not silently, they ought to fail with issues about byte-count.

> bumps the class version after adding the new member, ignoring the special semantics of Class-Version 0. The existing member fSorted will now be streamed.

This is indeed a bit quiet as the end result is likely that the newly read object would not behave (in some circumstances) as expected.

> That makes me think that Class-Version 0 is not very expressive for any class used directly or indirectly in IO. 

I agree that this is not a well know features ... especially since the actual semantic is not 'this class is transient but rather all the data member of this class are transient ... except the base classes!

I think we can help this by replacing (well superseding) the 0 by a constexpr unsigned short named kAllDataMembersTransient (okay that's too long of a name :( ) valued at 0 for backward compatibility.

Then in the report of your tool you could recommend to either switch to kAllDataMembersTransient or to switch to 1 + //!.

Cheers,
Philippe.